src/Cake.Core/Utilities/DisposableStopwatch.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics;
+
+using Cake.Core.Diagnostics;
+
+namespace Cake.Core.Utilities
+{
+    /// <summary>
+    /// Disposable Stopwatch used for timing how long a section of code takes and writing the result to the log.
+    /// </summary>
+    public class DisposableStopwatch : IDisposable
```

Comment:
Don't think we need a stopwatch at this point. Please open separate issue for reporting compilation time.

Reply:
Removed